java/simulated_ble/README.md

```diff
@@ -0,0 +1,306 @@
+
+The output is similar to the input, it could trigger hardware behavior (like the blinking LED), a message to other modules, or anything else (like printing to the console).
+
+Modules communicate with each other using `com.microsoft.azure.gateway.messaging.Message` class. The **Content** of a `Message` is a byte array which is capable of representing any kind of data you like. **Properties** is also available in `Message`, it is simply a string-to-string map. You may think of Properties as the headers in HTTP request, or the metadata of a file.
+
+In order to develop an IoT Edge module in Java, you need to create a new module class which extends from `com.microsoft.azure.gateway.core.GatewayModule`. Then you need to implement the abstract methods like `receive()` and `destroy()`. You can also choose to implement the optional methods like `start()` or `create()`. The following piece of code is a skeleton of writing a module.
+
+```java
+import com.microsoft.azure.gateway.core.Broker;
+import com.microsoft.azure.gateway.core.GatewayModule;
+import com.microsoft.azure.gateway.messaging.Message;
+
+public class MyEdgeModule extends GatewayModule {
+  public MyEdgeModule(long address, Broker broker, String configuration) {
+    /* Let the GatewayModule do the dirty work of initialization. It's also
+       a good time to parse your own configuration defined in Azure IoT Edge
+       configuration file (typically ./src/main/resources/gateway/gw-config.json) */
+    super(address, broker, configuration);
+  }
+
+  @Override
+  public void start() {
+    /* Acquire the resources you need. If you don't
+       need any resources, you may omit this method. */
+  }
+
+  @Override
+  public void destroy() {
+    /* It's time to release all resources. This method is required. */
+  }
+
+  @Override
+  public void receive(Message message) {
+    /* Logic to process the input message. This method is required. */
+    // ...
+    /* Use publish() method to do the output. You are
+       allowed to publish your new Message instance. */
+    this.publish(message);
+  }
+}
+```
+
+### Simulated BLE Module
+
+| Input | Processor | Output                   | Source File               |
+| ----- | --------- | ------------------------ | ------------------------- |
+| N/A   | N/A       | Temperature data message | `SimulatedBleModule.java` |
+
+This module simulates a [BLE](https://en.wikipedia.org/wiki/Bluetooth_Low_Energy) temperature sensor. It uses its own [Timer](https://docs.oracle.com/javase/8/docs/api/java/util/Timer.html) to send randomized temperature data at a fixed rate. It also includes a MAC Address and source type in the message properties.
+
+Both the rate and the MAC address are coming from the module configuration string (typically it is a JSON string). That's why we need to first parse the configuration in the constructor of our module:
```

Comment:
The values for the message rate and the MAC address are coming from the module configuration string argument of the method (typically a string in the format of a JSON object). In order to get the values from the string we need to first parse the configuration string in the constructor of our module: